3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/AnonIpEnrichmentSpec.scala

```diff
@@ -14,29 +14,101 @@ package com.snowplowanalytics.snowplow.enrich.common
+      "ipv6, anonymize 3"    !! "2605:2700:0:3:0:0:4713:93e3"   ! AnonIPv4Octets(3)       ! AnonIPv6Segments(3)     ! "2605:2700:0:3:0:x:x:x" |
+      "ipv6, anonymize 4"    !! "2605:2700:0:3::4713:93e3"      ! AnonIPv4Octets(4)       ! AnonIPv6Segments(4)     ! "2605:2700:0:3:x:x:x:x" |
+      "ipv6, anonymize 5"    !! "2605:2700:0:3::4713:93e3"      ! AnonIPv4Octets(4)       ! AnonIPv6Segments(5)     ! "2605:2700:0:x:x:x:x:x" |
+      "ipv6, anonymize 6"    !! "2605:2700:0:3::4713:93e3"      ! AnonIPv4Octets(4)       ! AnonIPv6Segments(6)     ! "2605:2700:x:x:x:x:x:x" |
+      "ipv6, anonymize 7"    !! "2605:2700:0:3::4713:93e3"      ! AnonIPv4Octets(4)       ! AnonIPv6Segments(7)     ! "2605:x:x:x:x:x:x:x" |
+      "ipv6, anonymize 8"    !! "2605:2700:0:3::4713:93e3"      ! AnonIPv4Octets(4)       ! AnonIPv6Segments(8)     ! "x:x:x:x:x:x:x:x" |
+      "ipv6, anonymize 8"    !! "2605:2700::4713:93e3"          ! AnonIPv4Octets(4)       ! AnonIPv6Segments(8)     ! "x:x:x:x:x:x:x:x" |
+      "ipv6, anonymize 8"    !! "2700::4713"                    ! AnonIPv4Octets(4)       ! AnonIPv6Segments(8)     ! "x:x:x:x:x:x:x:x" |
+      "ipv6, anonymize 8"    !! "2700:zzzz::gggg"               ! AnonIPv4Octets(4)       ! AnonIPv6Segments(6)     ! "2700:zzzz:x:x:x:x:x:x" |
+      "hybrid ipv6 + ipv4"   !! "f334::40cb:152.16.24.142"      ! AnonIPv4Octets(4)       ! AnonIPv6Segments(2)     ! "f334:0:0:0:0:40cb:x:x" |
+      "hybrid ipv6 + ipv4"   !! "f334::40cb:152.16.24.142"      ! AnonIPv4Octets(4)       ! AnonIPv6Segments(4)     ! "f334:0:0:0:x:x:x:x" |
+      "ipv4 mapped address"  !! "::FFFF:152.16.24.123"          ! AnonIPv4Octets(2)       ! AnonIPv6Segments(4)     ! "::FFFF:152.16.x.x" |
+      "ipv4 mapped address"  !! "::FFFF:152.16.24.123"          ! AnonIPv4Octets(4)       ! AnonIPv6Segments(4)     ! "::FFFF:x.x.x.x" |> {
+      (_, ip, octets, segments, expected) =>
+        AnonIpEnrichment(octets, segments).anonymizeIp(ip) must_== expected
     }
+
+  val ipv4Gen      = Gen.listOfN(4, Gen.choose(0, 255)).map(_.mkString("."))
+  val octetsNumGen = Gen.choose(1, 4)
+
+  val hexCharGen =
+    for {
+      num    <- Gen.numChar
+      hex    <- Gen.choose('a', 'f')
+      hexNum <- Gen.oneOf(num, hex)
+    } yield hexNum
+
+  val segmentGen = Gen.listOfN(4, hexCharGen).map(_.mkString)
+
+  //not shortened
+  val ipv6Gen        = Gen.listOfN(8, segmentGen).map(_.mkString(":"))
+  val segmentsNumGen = Gen.choose(1, 8)
+
+  val shortenedIPv6Gen =
+    (for {
+      segNum <- Gen.choose(2, 7)
+      ip     <- Gen.listOfN(segNum, segmentGen)
+    } yield ip.updated(segNum - 1, s":${ip(segNum - 1)}")).map(_.mkString(":"))
+
+  def e2 =
+    forAll(ipv4Gen, octetsNumGen) {
+      case (ip, octetsNum) =>
+        val anon = AnonIpEnrichment(AnonIPv4Octets(octetsNum), AnonIPv6Segments(1)).anonymizeIp(ip)
+        anon.count(_ == 'x') ==== octetsNum
```

Comment:
Are we sure these are assertion-based test?

E.g. usually specs2 immutable test with

```scala
1 must beEqualTo(2)
"ok" must beEqualTo("ok")
```

Won't fail because its value is `ok` and first statement will be simply ignored.

Reply:
Great catch @chuwy! Thank you. I haven't been using specs2 immutable spec before.